### PR TITLE
docs: remove mentions of scan-build

### PR DESCRIPTION
**Pull request by dundargoc**
These same checks are already checked by the clang-analyzer job, so this
duplication is not necessary.

### CODE REVIEW

**Comment by dundargoc** [comment]
This takes too long. Can't split this up into multiple jobs, building neovim takes too long. Wait for upload-articat and download-artifact v4 and retry this again.

**Comment by dundargoc** [comment]
TODO: write TODOs that will make sense when I read after a while

File: src/nvim/CMakeLists.txt [hunk 1]
```diff
@@ -852,6 +852,42 @@ add_glob_target(
...
+    '
+    EXCLUDE ${EXCLUDE_CLANG_TIDY} ${ARG_EXCLUDE_FILES})
+
+  add_dependencies(clang-analyzer ${ARG_CHECK})
+  add_dependencies(${ARG_CHECK} copy_compile_commands)
+endfunction()
+
+# TODO(dundargoc): remove exclude files after fixing the corresponding error
+clang_analyzer(CHECK clang-analyzer-core.UndefinedBinaryOperatorResult
+  EXCLUDE_FILES memfile.c marktree.c)
```
Comment (zeertzjq) [inline src/nvim/CMakeLists.txt:883]: Where does memfile.c have this warning?

Comment (dundargoc) [reply src/nvim/CMakeLists.txt]: Probably an old warning that got removed at some point. Removed, thanks.

**Comment by dundargoc** [comment]
Need to merge https://github.com/neovim/doc/pull/58 first

File: src/nvim/CMakeLists.txt [hunk 2]
```diff
@@ -836,6 +836,9 @@ add_glob_target(
 # These are the same warnings as https://neovim.io/doc/reports/clang/. The
```
Comment (justinmk) [inline src/nvim/CMakeLists.txt:836]:
```suggestion
# The
```

**Comment by dundargoc** [comment]
Unsure if I should mention `make clang-analyzer` in the place where `scan-build` used to be or if it's obvious?

**Comment by dundargoc** [comment]
Todo: need to remove link from readme

**Review by justinmk — APPROVED** [review]

**Pull request merged**